scanpipe/pipes/__init__.py

```diff
@@ -182,7 +192,7 @@ def update_or_create_package(project, package_data, codebase_resources=None):
     else:
         package = DiscoveredPackage.create_from_data(project, package_data)
 
-    if codebase_resources:
+    if not is_empty_queryset(codebase_resources):
```

Comment:
@tdruez When I was testing out the purldb matching portion of the d2d pipeline, I timed out on this line:
```
Traceback:
  File "/opt/scancodeio/scanpipe/pipelines/__init__.py", line 120, in execute
    step(self)
  File "/opt/scancodeio/scanpipe/pipelines/deploy_to_develop.py", line 154, in match_purldb
    d2d.match_purldb_resources(
  File "/opt/scancodeio/scanpipe/pipes/d2d.py", line 669, in match_purldb_resources
    matched_count += matcher_func(
                     ^^^^^^^^^^^^^
  File "/opt/scancodeio/scanpipe/pipes/d2d.py", line 559, in match_purldb_package
    _, matched_resources_count = create_package_from_purldb_data(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/scanpipe/pipes/d2d.py", line 520, in create_package_from_purldb_data
    package = pipes.update_or_create_package(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/scanpipe/pipes/__init__.py", line 143, in update_or_create_package
    if codebase_resources:
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/query.py", line 412, in __bool__
    self._fetch_all()
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1595, in execute_sql
    return list(result)
           ^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 2093, in cursor_iter
    for rows in iter((lambda: cursor.fetchmany(itersize)), sentinel):
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 2093, in <lambda>
    for rows in iter((lambda: cursor.fetchmany(itersize)), sentinel):
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/django/db/utils.py", line 98, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/psycopg/cursor.py", line 847, in fetchmany
    records = self._tx.load_rows(
              ^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/psycopg/_transform.py", line 307, in load_rows
    val = res.get_value(row, col)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/psycopg/pq/pq_ctypes.py", line 818, in get_value
    length: int = impl.PQgetlength(self._pgresult_ptr, row_number, column_number)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/scancodeio/venv/lib/python3.11/site-packages/rq/timeouts.py", line 63, in handle_death_penalty
    raise self._exception('Task exceeded maximum timeout value ({0} seconds)'.format(self._timeout))
```
    
    
This was due to a bug where I set `codebase_resources` to have all of the CodebaseResources from the `to/` codebase, which was about 550,000 in number. 
    
> Why do we need a custom function to handle this? Do we have a unit test that shows the issue using directly the codebase_resources var?
    
I made this logic into its own function because I do a similar check here: https://github.com/nexB/scancode.io/blob/7eb5d4603a0c625e19507053a0ec8714243689b5/scanpipe/pipes/d2d.py#L427